hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -102,27 +124,237 @@
+				validateFalse(newBalanceIsInvalid, TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
+			}
+		}
+	}
+
+	private void updateAdminKeyIfAppropriate(final TokenUpdateTransactionBody changes) {
+		if (changes.hasAdminKey()) {
+			var newAdminKey = changes.getAdminKey();
+			if (REMOVES_ADMIN_KEY.test(newAdminKey)) {
+				this.setAdminKey(UNUSED_KEY);
+			} else {
+				this.setAdminKey(asFcKeyUnchecked(changes.getAdminKey()));
+			}
+		}
+	}
+
+	private void updateAutoRenewAccountIfAppropriate(final Account newAutoRenewAccount) {
+		if (autoRenewAccount != null) {
+			this.setAutoRenewAccount(newAutoRenewAccount);
+		}
+	}
+
+	private void updateAutoRenewPeriodIfAppropriate(final TokenUpdateTransactionBody changes) {
+		if (this.hasAutoRenewAccount()) {
+			final var changedAutoRenewPeriod = changes.getAutoRenewPeriod().getSeconds();
+			if (changedAutoRenewPeriod > 0L) {
+				this.setAutoRenewPeriod(changedAutoRenewPeriod);
+			}
+		}
+	}
+
+	private void updateKeyOfTypeIfAppropriate(boolean hasKey, Supplier<Key> supplier, Consumer<JKey> consumer) {
+		if (hasKey) {
+			consumer.accept(asFcKeyUnchecked(supplier.get()));
+		}
+	}
+
+	private void updateSymbolIfAppropriate(final TokenUpdateTransactionBody changes) {
+		if (changes.getSymbol().length() > 0) {
+			this.setSymbol(changes.getSymbol());
+		}
+	}
+
+	private void updateNameIfAppropriate(final TokenUpdateTransactionBody changes) {
+		if (changes.getName().length() > 0) {
+			this.setName(changes.getName());
+		}
+	}
+
+	private void updateMemoIfAppropriate(final TokenUpdateTransactionBody changes) {
+		if (changes.hasMemo()) {
+			this.setMemo(changes.getMemo().getValue());
+		}
+	}
+
+	private void updateExpiryIfAppropriate(final TokenUpdateTransactionBody changes) {
+		var expiry = changes.getExpiry().getSeconds();
+		if (expiry != 0L) {
+			this.setExpiry(expiry);
+		}
+	}
+
+	private void updateTreasuryIfAppropriate(final Account newTreasury) {
+		if (newTreasury != null) {
+			setTreasury(newTreasury);
+			setHasUpdatedTreasury(true);
+		}
+	}
+
 	/**
 	 * Creates a new instance of the model token, which is later persisted in state.
 	 *
-	 * @param tokenId
-	 * 		the new token id
-	 * @param op
-	 * 		the transaction body containing the necessary data for token creation
-	 * @param treasury
-	 * 		treasury of the token
-	 * @param autoRenewAccount
-	 * 		optional(nullable) account used for auto-renewal
-	 * @param customFees
-	 * 		a list of valid custom fees
-	 * @param consensusTimestamp
-	 * 		the consensus time of the token create transaction
+	 * @param tokenId            the new token id
+	 * @param op                 the transaction body containing the necessary data for token creation
+	 * @param treasury           treasury of the token
+	 * @param autoRenewAccount   optional(nullable) account used for auto-renewal
+	 * @param customFees         a list of valid custom fees
+	 * @param consensusTimestamp the consensus time of the token create transaction
```

Comment:
You can revert those formatting changes. Please note that this comment **applies to all of those changes**